packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4497,66 +4497,452 @@ class PositionedDirectional extends StatelessWidget {
+///      laid out is 200 pixels.
+///    - There is one [Expanded] with a [flex] of 1.
+///    - There is one [Flexible] with a [flex] of 1.
+///    - The flexes are equal, so half of the space (100 pixels) is reserved
+///      for each of them.
+///    - The [Expanded] and [Flexible] each has a child that is 50 pixels
+///      high.
+///    - The [Expanded] will pass its child tight constraints with a maximum
+///      size of 100 pixels, what was reserved for it. The
+///      fact that it's passing in tight constraints causes the child to be
+///      forced to be 100 pixels high, because the tight constraints cause it
+///      to override the [Expanded]'s child's height parameter of 50 pixels.
+///    - The [Flexible] passes loose constraints to its child, in this case
+///      a minimum of 0 pixels and maximum of 100 pixels, which was the space
+///      reserved for it. The child's height parameter is set to 50
+///      pixels, and the loose constraints will allow the [Flexible]'s child
+///      to be 50 pixels high.
+///
+///    When you look at the rendered result, it seems as though the [Column]
+///    is short by 50 pixels because all the children you can see add up to
+///    only 250 pixels. However, the [Column] is not short, it is the full
+///    300 pixels high. What is happening is the [Column] is 300 pixels high
+///    but all of its children only add up to 250 pixels high. The [Expanded]
+///    _did_ use all of the space it was offered, but it was only offered 100
+///    pixels. The reason the [Column] looks short is that it has a transparent
+///    background, and you're looking _through_ the part that wasn't used.
+///    If a [Flex] had borders then this would be easy to see, but it doesn't.
+///
+/// 4. _In most cases_ the final size of the [Flex] in its cross axis will be
+///    the same as its largest child in the cross axis, but only as long
+///    as that child's size in the cross axis will fit within the maximum
+///    cross axis constraint set by the parent of the [Flex]. The one case
+///    in which the final size of the [Flex] will not be the same size as its
+///    largest child is if you set the [crossAxisAlignment] parameter to
+///    [CrossAxisAlignment.baseline] and the maximum cross axis constraint
+///    is more than the cross axis size of the largest child. More on this
+///    in step 6.
+///
+///    If the largest child in the cross axis doesn't have a set
+///    size and instead just wants to be as large as it's allowed to be there
+///    are two possible outcomes. If the parent of the [Flex] has set a
+///    maximum constraint in the cross axis, then the [Flex] will be that
+///    size. If the parent of the [Flex] has not set any maximum size and a
+///    child tries to be as large as possible in the cross axis, it is
+///    impossible to calculate a final size and you may get the error:
+///
+///    "BoxConstraints forces an infinite (height or width)".
```

Comment:
Again, you might add something about how it can be fixed.

Reply:
Handling this offline in the IDE.